source/grid/tria.cc

```diff
@@ -2157,1539 +2158,421 @@ namespace internal
-                  // face_orientation=true, face_flip=false, face_rotation=true
-                  test_quad_5({quad[2], quad[3], quad[1], quad[0]}),
-                  // face_orientation=true, face_flip=true, face_rotation=false
-                  test_quad_6({quad[1], quad[0], quad[3], quad[2]}),
-                  // face_orientation=true, face_flip=true, face_rotation=true
-                  test_quad_7({quad[3], quad[2], quad[0], quad[1]});
-
-                if (needed_quads.find(test_quad_1) == needed_quads.end() &&
-                    needed_quads.find(test_quad_2) == needed_quads.end() &&
-                    needed_quads.find(test_quad_3) == needed_quads.end() &&
-                    needed_quads.find(test_quad_4) == needed_quads.end() &&
-                    needed_quads.find(test_quad_5) == needed_quads.end() &&
-                    needed_quads.find(test_quad_6) == needed_quads.end() &&
-                    needed_quads.find(test_quad_7) == needed_quads.end())
-                  needed_quads[quad] =
-                    std::make_pair(triangulation.end_quad(), orientation);
-              }
+            // quad line orientations
+            faces.quads_line_orientations.assign(size * faces_per_cell, -1);
           }
+      }
 
 
-        /////////////////////////////////
-        // enter the resulting quads into
-        // the arrays of the Triangulation
-        //
-        // first reserve enough space
-        reserve_space(*triangulation.faces, 0, needed_quads.size());
-        reserve_space(triangulation.faces->quads, 0, needed_quads.size());
-
-        {
-          typename Triangulation<dim, spacedim>::raw_quad_iterator quad =
-            triangulation.begin_raw_quad();
-          typename std::map<
-            std::vector<int>,
-            std::pair<typename Triangulation<dim, spacedim>::quad_iterator,
-                      std::array<bool, GeometryInfo<dim>::lines_per_face>>,
-            QuadComparator>::iterator q;
-          for (q = needed_quads.begin(); quad != triangulation.end_quad();
-               ++quad, ++q)
-            {
-              quad->set_bounding_object_indices(
-                {q->first[0], q->first[1], q->first[2], q->first[3]});
-              quad->set_used_flag();
-              quad->clear_user_flag();
-              quad->clear_user_data();
-              // set the line orientation
-              quad->set_line_orientation(0, q->second.second[0]);
-              quad->set_line_orientation(1, q->second.second[1]);
-              quad->set_line_orientation(2, q->second.second[2]);
-              quad->set_line_orientation(3, q->second.second[3]);
-
-
-              // now set the iterator for
-              // this quad
-              q->second.first = quad;
-            }
-        }
 
-        /////////////////////////////////
-        // finally create the cells
-        reserve_space(triangulation.levels[0]->cells, cells.size());
+      /**
+       * TODO: use the other one
```

Comment:
?

Reply:
I think I might keep these functions (with a different name). They are simple and specialized for the coarse grid. What is you opinion?